### PR TITLE
New value source tests

### PR DESCRIPTION
**Pull request by mlopatkin**
The test case for #23689 and some more.

### CODE REVIEW

File: subprojects/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/ConfigurationCacheValueSourceIntegrationTest.groovy [hunk 1]
```diff
@@ -367,4 +368,160 @@ class ConfigurationCacheValueSourceIntegrationTest extends AbstractConfiguration
...
+        when:
+        configurationCacheRun()
+
+        then:
+        configurationCache.assertStateLoaded()
+        outputContains("execution value = someValue")
+    }
+
+    @ToBeImplemented("https://github.com/gradle/gradle/issues/23689")
+    def "value source can read mutated system property at configuration time with Java API"() {
```
Comment (bamboo) [inline subprojects/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/ConfigurationCacheValueSourceIntegrationTest.groovy:422]: 🧪 Perhaps a similar test for environment variables?

Comment (mlopatkin) [reply subprojects/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/ConfigurationCacheValueSourceIntegrationTest.groovy:422]: >Perhaps a similar test for environment variables?

There's no API to alter environment variables at runtime without using unforgivable curses, so I don't think the test for it is valuable enough.

Comment (bamboo) [reply subprojects/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/ConfigurationCacheValueSourceIntegrationTest.groovy:422]: Ah, right.

**Review by bamboo — APPROVED** [review]

**Comment by mlopatkin** [comment]
@bot-gradle test and merge

**Pull request merged**